liberty-ls/src/main/java/io/openliberty/tools/langserver/diagnostic/LibertyPropertiesDiagnosticService.java

```diff
@@ -82,30 +82,36 @@ private List<Diagnostic> computeInvalidValuesDiagnostic(PropertiesValidationResu
             
             // Currently the last arg (getIntegerRange) is only used for the Integer messages which use {2}. Otherwise null is passed and is ignored by the other messages.
             String message = MessageFormat.format(messageTemplate, validationResult.getValue(), property, ServerPropertyValues.getIntegerRange(property));
-            lspDiagnostics.add(new Diagnostic(computeRange(validationResult, lineContentInError, validationResult.getValue()), message));
+            lspDiagnostics.add(new Diagnostic(computeRange(validationResult, lineContentInError), message));
         }
         return lspDiagnostics;
     }
 
     /**
-     * Finds the given string in the entry line, returns Range containing position information for the given string.
+     * Returns a Range containing position information for the diagnostic
      * @param validationResult
      * @param lineContentInError - Line entry from property file.
-     * @param value - String to look for to highlight for diagnostic.
      * @return
      */
-    private Range computeRange(PropertiesValidationResult validationResult, String lineContentInError, String value) {
-        int startCharacter, endCharacter;
-        int indexOfValue = lineContentInError.lastIndexOf(value);
-        if (indexOfValue != -1) {
-            startCharacter = indexOfValue;
-            endCharacter = indexOfValue + value.length();
-        } else {
-            startCharacter = 0;
-            endCharacter = lineContentInError.length();
-        }
+    private Range computeRange(PropertiesValidationResult validationResult, String lineContentInError) {
+        int equalIndex = lineContentInError.indexOf("=");
         int lineNumber = validationResult.getLineNumber();
-        return new Range(new Position(lineNumber, startCharacter), new Position(lineNumber, endCharacter));
+        Integer startChar = validationResult.getStartChar();
+        Integer endChar = validationResult.getEndChar();
+        // use range if given
+        if (startChar != null && endChar != null) {
+            return new Range(new Position(lineNumber, startChar), new Position(lineNumber, endChar));
+        }
+
+        // otherwise calculate range
+        if (startChar != null) {
+            // currently unused. use either whole line or equal index
+            endChar = lineContentInError.length();
+        } else if (endChar != null) {
+            startChar = equalIndex + 1;
```

Comment:
I feel like lines 101 through 112 could be simplified to something like this:

```
Position start = startChar == null ? new Position(lineNumber, equalIndex +1) : new Position(lineNumber, startChar);
Position end = endChar == null ? new Position(lineNumber, lineContentInError.length()) : new Position(lineNumber, endChar);
return new Range(start, end);
```

I wasn't sure what it was supposed to do if both `startChar` and `endChar` were null? Is that possible? Your code did not handle that exactly. It would fall through and use the nulls I think.